### PR TITLE
22 ensure correct datums

### PR DESCRIPTION
**Pull request by rosepearson**
No consideration has been given to vertical and horizontal datums for input data and even the output DEM. Begin adding support for vertical datums by first adding it for LiDAR data. _There are instances where LiDAR data has a vertical datum, but it has not been encoded in the LAZ file. As a result, we will need to support specifying the datum of the DEM we are generating, as well as optionally specifying the datum of the data we are reading - in the case it may not be specified._

- [X] Add an the concept of an overall vertical and horizontal datum (i.e. defines the datum of the output DEM)
- [X] Add an optional vertical and/or horizontal datum for LiDAR datasets to apply to all LiDAR files read in for that dataset
- [x] Apply horizontal and vertical datums re-projections to each LiDAR dataset
  - [x] If a LiDAR dataset horizontal/vertical datum is specified - then define as IN_SRS to override the CRS in the LAZ files
- [X] Update the tests accordingly
- [x] Update the documentation

Things that will be done in a later issue/PR include:
* Add a test using a local datum
* Add support for vertical and horizontal datum's in vectors

### CODE REVIEW

**Comment by rosepearson** [comment]
Hi @AliceHarang, I've made the changes I need to ensure vertical and horizontal datum information is considered for LAZ files using PDAL. There are instances where the vertical information is not encoded in the file, but it is in that datum - which is why I've allowed that information to be specified in the instruction file.
A code review would be great!

File: src/main.py [hunk 1]
```diff
@@ -31,8 +32,12 @@ def launch_processor(args):
         instructions = json.load(file_pointer)
 
     # run the pipeline
+    start_time = time.time()
     runner = processor.GeoFabricsGenerator(instructions)
     runner.run()
+    end_time = time.time()
+
+    print(f"Execution time is {end_time - start_time}")
```
Comment (AliceHarang) [inline src/main.py:40]: Good idea to add a timing!

File: src/geofabrics/processor.py [hunk 2]
```diff
@@ -122,40 +146,83 @@ def get_vector_paths(self, key) -> list:
...
+                    shutil.make_archive(base_name=vector_dir, format='zip', root_dir=vector_dir)
+                    shutil.rmtree(vector_dir)
+                    paths.append(layer_dir / f"{key}.zip")
         return paths
 
-    def get_lidar_file_list(self, verbose) -> list:
-        """ Load or construct a list of lidar tiles to construct a DEM from. """
+    def get_lidar_dataset_crs(self, data_service, dataset_name, verbose: bool = False) -> dict:
+        """ Checks to see if source CRS of an associated LiDAR datasets has be specified in the instruction file. If it
+        has been specified, this CRS is returned, and will later by used to override the CRS encoded in the LAS files.
```
Comment (AliceHarang) [inline src/geofabrics/processor.py:188]: typo: by=>be

File: src/geofabrics/processor.py [hunk 3]
```diff
@@ -122,40 +146,83 @@ def get_vector_paths(self, key) -> list:
...
+        data_services = {"linz": geoapis.vector.Linz, "lris": geoapis.vector.Lris}  # API name and geoapis class pairs
+        for data_service in data_services.keys():
+            if self.check_apis(data_service) and key in self.instructions['instructions']['apis'][data_service]:
+                cache_dir = pathlib.Path(self.get_instruction_path('local_cache'))
+                api_key = self.instructions['instructions']['apis'][data_service]['key']
+
+                assert self.check_instruction_path('local_cache'), "Local cache file path must exist to specify the" + \
+                    f" location to download vector data from the vector APIs: {data_services}"
+                assert self.catchment_geometry is not None, "The `self.catchment_directory` object must exist " + \
+                    "before avector is downloaded using `vector.Linz`"
```
Comment (AliceHarang) [inline src/geofabrics/processor.py]: Add some comment to explain your handling of apis keys here

**Comment by AliceHarang** [comment]
Hi Rose, 
It seems good to me. 
The only point would be to try to avoid/clarify the acronyms you are using, specially for the code output to user (CRS for example).
(EPSG is ok for me, but you can still clarify by saying "the coordinate system EPSG", or something more appropriate).
Apparrt for this, it is all good for me!

**Comment by rosepearson** [comment]
> Hi Rose,
> It seems good to me.
> The only point would be to try to avoid/clarify the acronyms you are using, specially for the code output to user (CRS for example).
> (EPSG is ok for me, but you can still clarify by saying "the coordinate system EPSG", or something more appropriate).
> Apparrt for this, it is all good for me!

Awesome. Thanks for that feedback! Yup - I'll go through and update all the printouts as you suggest to make it clearer.

**Review by AliceHarang — APPROVED** [review]

File: src/geofabrics/processor.py (hunk 3 above)
Comment (rosepearson) [reply src/geofabrics/processor.py]: Done - in next commit :)

File: src/geofabrics/processor.py (hunk 2 above)
Comment (rosepearson) [reply src/geofabrics/processor.py:188]: Done - in next commit :)

**Pull request merged**